cmd/maya-exporter/app/collector/mock/mock.go

```diff
@@ -0,0 +1,43 @@
+package mock
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// PrometheusHandler is used to mock the behaviour of prometheus's client-go
+// apis for testing purpose.
+// Here, it registers the collector and then starts a test http server and send
+// a get request to which server reply with the response as expected
+func PrometheusHandler(t *testing.T, col prometheus.Collector) ([]byte, *httptest.Server) {
```

Comment:
Avoid using `testing` as an argument in these functions which are helpers
IMO we can avoid use of `testing` package altogether

Reply:
Can we rename this to `PrometheusService`
The caller code will make use of this as `mock.PrometheusService(..)`

Reply:
Sounds good